extensions-core/hdfs-storage/src/main/java/io/druid/storage/hdfs/HdfsStorageDruidModule.java

```diff
@@ -126,5 +129,7 @@ public void configure(Binder binder)
     Binders.taskLogsBinder(binder).addBinding("hdfs").to(HdfsTaskLogs.class);
     JsonConfigProvider.bind(binder, "druid.indexer.logs", HdfsTaskLogsConfig.class);
     binder.bind(HdfsTaskLogs.class).in(LazySingleton.class);
+    LifecycleModule.register(binder, HdfsStorageAuthentication.class);
```

Comment:
@b-slim Another way to register a hook with lifecycle is by registering a Handler with Lifecycle using lifecycle.addMaybeStartHandler. I believe that should work with extension as well. Refer SQLAuditManagerProvider for example